lib/ansible/modules/commands/expect.py

```diff
@@ -193,6 +197,9 @@ def main():
             # Use pexpect.runu in pexpect>=3.3,<4
             out, rc = pexpect.runu(args, timeout=timeout, withexitstatus=True,
                                    events=events, cwd=chdir, echo=echo)
+        finally:
+            if PY2:
+                sys.exc_clear()  # Avoid false positive traceback in fail_json() on Python 2
```

Comment:
This one looks good but should be part of the TypeError block instead of in a finally block.  The reason to me is that if the TypeError block runs successfully, then the code has successfully fallen back to code which handled the error.  If the TypeError block throws its own exception (or the initial pexpect.run() failed for a different reason than TypeError) then we haven't handled the exception and the traceback could be helpful.

Reply:
You are right. My main reason for moving it to `finally` was that I did have to modify the existing handling. So I could leave `pass` where it was stated.